File: backend/src/main/java/ru/bowling/bowlingapp/DTO/ClubSummaryDTO.java
package ru.bowling.bowlingapp.DTO;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ClubSummaryDTO {
    private Long id;
    private String name;
    private String address;
    private Integer lanesCount;
    private String contactPhone;
    private String contactEmail;
}

File: frontend/lib/models/club_summary_dto.dart
class ClubSummaryDto {
  final int id;
  final String name;
  final String? address;
  final int? lanesCount;
  final String? contactPhone;
  final String? contactEmail;

  ClubSummaryDto({
    required this.id,
    required this.name,
    this.address,
    this.lanesCount,
    this.contactPhone,
    this.contactEmail,
  });

  factory ClubSummaryDto.fromJson(Map<String, dynamic> json) {
    return ClubSummaryDto(
      id: (json['id'] as num).toInt(),
      name: json['name'] as String? ?? '',
      address: json['address'] as String?,
      lanesCount: (json['lanesCount'] as num?)?.toInt(),
      contactPhone: json['contactPhone'] as String?,
      contactEmail: json['contactEmail'] as String?,
    );
  }
}
